### ISSUE TITLE
x4 training don't save weights, only logs

### ISSUE THREAD

**Issue by Wazted** [idealo/image-super-resolution#89]
Hello , I wanted to train the SRGAN for x4 upscaling but after my 80 epochs I went to the weights folder but there were only training session logs.

```
.:
rdn-C6-D20-G64-G064-x2_ArtefactCancelling_epoch219.hdf5  rrdn-C4-D3-G64-G064-T10-x2  rrdn-C4-D3-G64-G064-T10-x4

./rrdn-C4-D3-G64-G064-T10-x2:
2019-11-14_16:07

./rrdn-C4-D3-G64-G064-T10-x2/2019-11-14_16:07:
session_config.yml

./rrdn-C4-D3-G64-G064-T10-x4:
2019-11-13_12:09  2019-11-13_13:47  2019-11-13_13:48  2019-11-14_15:11

./rrdn-C4-D3-G64-G064-T10-x4/2019-11-13_12:09:
session_config.yml

./rrdn-C4-D3-G64-G064-T10-x4/2019-11-13_13:47:
session_config.yml

./rrdn-C4-D3-G64-G064-T10-x4/2019-11-13_13:48:
session_config.yml

./rrdn-C4-D3-G64-G064-T10-x4/2019-11-14_15:11:
session_config.yml
```
This is the `ls -R` in the weights directory, I did two train one x2 and one x4

````
from ISR.models import RRDN
from ISR.models import Discriminator
from ISR.models import Cut_VGG19
from ISR.train import Trainer

lr_train_patch_size = 40
layers_to_extract = [5, 9]
scale = 4
hr_train_patch_size = lr_train_patch_size * scale

rrdn  = RRDN(arch_params={'C':4, 'D':3, 'G':64, 'G0':64, 'T':10, 'x':scale}, patch_size=lr_train_patch_size)
f_ext = Cut_VGG19(patch_size=hr_train_patch_size, layers_to_extract=layers_to_extract)
discr = Discriminator(patch_size=hr_train_patch_size, kernel_size=3)

loss_weights = {
  'generator': 0.0,
  'feature_extractor': 0.0833,
  'discriminator': 0.01
}
losses = {
  'generator': 'mae',
  'feature_extractor': 'mse',
  'discriminator': 'binary_crossentropy'
}

log_dirs = {'logs': './logs', 'weights': './weights'}

learning_rate = {'initial_value': 0.0004, 'decay_factor': 0.5, 'decay_frequency': 30}

flatness = {'min': 0.0, 'max': 0.15, 'increase': 0.01, 'increase_frequency': 5}

trainer = Trainer(
    generator=rrdn,
    discriminator=discr,
    feature_extractor=f_ext,
    lr_train_dir='low_res/training/images',
    hr_train_dir='high_res/training/images',
    lr_valid_dir='low_res/validation/images',
    hr_valid_dir='high_res/validation/images',
    loss_weights=loss_weights,
    learning_rate=learning_rate,
    flatness=flatness,
    dataname='image_dataset',
    log_dirs=log_dirs,
    weights_generator=None,
    weights_discriminator=None,
    n_validation=40,
)

trainer.train(
    epochs=80,
    steps_per_epoch=500,
    batch_size=16,
    monitored_metrics={'val_PSNR_Y': 'max'}
)
````
Here you have my code.

If you see something wrong tell me it will be really helpfull.

_I love the idea of image super resolution !_

**Comment by charlnaya** [comment]
try using the following for metrics, that worked for me. 

`monitored_metrics={'val_generator_PSNR_Y': 'max'}`

**Comment by cfrancesco** [comment]
yeah, @charlnaya suggestion is most likely to be correct.
Keras will internally change the name of the metric if there is more than one model, so it will be 'val_generator_PSNR_Y' if using GANS, other just 'val_PSNR_Y', disregarding the model name.
The colab notebook def need to be updated.
